test/Base.t.sol

```diff
@@ -76,7 +88,7 @@ abstract contract Base_Test is Assertions, Calculations, DeployOptimized, Modifi
         // Deploy the NFT descriptor mock.
         nftDescriptorMock = new NFTDescriptorMock();
 
-        // Create users for testing.
+        // Create users for testing. Note that this has to go after the protocol is deployed due to ERC-20 approvals.
```

Comment:
Not sure what you meant here. Why does it have to go after deployment?

PS: for to do tasks, we add `TODO` and use [TODO Tree](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree) to identify them.

Reply:
I've tried to move that code before the `deployProtocolConditionally` function, and I got an error.

The reason was that the user creation function assumes the Lockup contracts to be deployed.